x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ApplicationPrivilege.java

```diff
@@ -192,18 +193,15 @@ public static Set<ApplicationPrivilege> get(String application, Set<String> name
         if (name.isEmpty()) {
             return Collections.singleton(NONE.apply(application));
         } else if (application.contains("*")) {
-            Predicate<String> predicate = Automatons.predicate(application);
-            final Set<ApplicationPrivilege> result = stored.stream()
+            final Set<ApplicationPrivilege> result = Sets.newHashSet(resolve(application, name, Collections.emptyMap()));
```

Comment:
This is the real part of the change. 
Previously this `resolve` would only happen if there were no direct matches.
Now we always include the resolved privilege as well as the matches.